Code/GraphMol/MMPA/Wrap/rdMMPA.cpp

```diff
@@ -47,6 +48,116 @@ python::tuple fragmentMolHelper(const RDKit::ROMol& mol, unsigned int maxCuts,
   }
   return python::tuple(pyres);
 }
+
+python::tuple fragmentMolHelper2(const RDKit::ROMol& mol,
+                                 unsigned int minCuts,
+                                 unsigned int maxCuts,
+                                 unsigned int maxCutBonds,
+                                 const std::string& pattern,
+                                 bool resultsAsMols) {
+  std::vector<std::pair<RDKit::ROMOL_SPTR, RDKit::ROMOL_SPTR> > tres;
+  bool ok = RDKit::MMPA::fragmentMol(mol, tres, minCuts, maxCuts, maxCutBonds,
+                                     pattern);
+  python::list pyres;
+  if (ok) {
+    for (std::vector<std::pair<RDKit::ROMOL_SPTR,
+                               RDKit::ROMOL_SPTR> >::const_iterator pr =
+             tres.begin();
+         pr != tres.end(); ++pr) {
+      python::list lres;
+      if (resultsAsMols) {
+        lres.append(pr->first);
+        lres.append(pr->second);
+      } else {
+        if (pr->first) {
+          lres.append(RDKit::MolToSmiles(*(pr->first), true));
+        } else {
+          lres.append("");
+        }
+        lres.append(RDKit::MolToSmiles(*(pr->second), true));
+      }
+      pyres.append(python::tuple(lres));
+    }
+  }
+  return python::tuple(pyres);
+}
+
+template<class T>
+std::vector<unsigned int> ConvertToVect(T bbs) {
+  std::vector<unsigned int> vect;
+  size_t num = python::extract<unsigned int>(bbs.attr("__len__")());
+  vect.reserve(num);
+  for(size_t i=0; i<num; ++i) {
+    vect.push_back(python::extract<unsigned int>(bbs[i]));
+  }
+  return vect;
+}
+
+python::tuple fragmentMolHelper3(const RDKit::ROMol& mol,
```

Comment:
I think you can combine this and `fragmentMolHelper4()` by changing the type of the `ob` argument to `python::object` and using `pythonObjectToVect()`, that would allow anything that supports `__len__` and `__getitem__` to be used.